implementation/src/main/java/io/smallrye/config/SmallRyeConfig.java

```diff
@@ -327,8 +332,25 @@ public boolean rawValueEquals(String name, String expected) {
         return Objects.equals(expected, getRawValue(name));
     }
 
-    @Experimental("Extension to the original ConfigSource to allow retrieval of additional metadata on config lookup")
     public ConfigValue getConfigValue(String name) {
+        try {
+            return getRawConfigValue(name);
+        } catch (NoSuchElementException e) {
+            return ConfigValue.builder().withName(name).build();
+        }
+    }
+
+    /**
+     * Return the {@link org.eclipse.microprofile.config.ConfigValue} for the specified property name from the underlying
+     * {@linkplain ConfigSource
+     * configuration source}. The lookup of the configuration is performed immediately, meaning that calls to
+     * {@link org.eclipse.microprofile.config.ConfigValue} will always yield the same results.
+     *
+     * @param name The configuration property name
+     * @return the resolved property value as a {@link org.eclipse.microprofile.config.ConfigValue}
+     * @throws NoSuchElementException in case the property value could not be expanded.
+     */
+    public ConfigValue getRawConfigValue(String name) {
```

Comment:
I think it is confusing to have two public API's `getConfigValue` and `getRawConfigValue`. Can we use only one?

Reply:
You mean, you would like me to overload `getConfigValue` and adds a new method of type `getConfigValue(String name, boolean noSuchElementExceptionForbidden)`, is that what you had in mind?